### ISSUE TITLE
Document interesting `tmt` design patterns

### ISSUE THREAD

**Issue by LecrisUT** [psss/tmt#2822]
After designing a few tmt plans, I have found a few design patterns, and I think it would be good to document and give ideas for the new users what can be done, and how they can use other project's tmt framework:

#### Separating `.distro` with its own `fmf` root from the main project's tests

This is helpful for synchronizing the tests with downstream packaging and perform common distro specific tests, e.g. `rpminspect`, `rpmlint`, etc.
- `testing-farm` specific test that are run after `copr-build` are pointing here, where `plan.import` is used to import the main tests
- Main project's fmf root (I prefer to have it in `tests` since many projects want to avoid adding files to the root), are designed such that they can be used by all frameworks: `testing-farm`, `local`, `github-action`, etc.

#### Providing integration tests with upstream dependencies

Within the integration tests, tag the tests with the appropriate dependency (+ `integration`). That way within your CI, you can checkout various versions of dependencies, e.g. `main` branch, and test the integration regularly. Upstream can also then call the respective integration tests to ensure/announce compatibility breakage. 

#### Providing integration tests for upstream users

Add plans that can be used by upstream projects to ensure they are not creating any breaking changes. This is the inverse of the previous point, where you want upstream to run some tests to ensure you are not breaking anything. Could be smoke tests, integration tests and functional tests.

#### Other common test organization

I like to organize my tests under the common `unit`, `integration`, `functional`, but also `package`, `smoke`, `example`.

**Comment by psss** [comment]
Yeah, definitely makes sense to extend the documentation with more inspirative examples of how things can be efficiently done. For the integration part I've recently filed this:

* #4028 

Adding it as a subissue of this one.
